cmd/dagger/listen.go

```diff
@@ -27,7 +27,7 @@ func Listen(cmd *cobra.Command, args []string) {
 	ctx := context.Background()
 	if err := withEngine(ctx, "", func(ctx context.Context, r *router.Router) error {
 		fmt.Fprintf(os.Stderr, "==> server listening on http://%s/query\n", listenAddress)
-		return http.ListenAndServe(listenAddress, r)
+		return http.ListenAndServe(listenAddress, r) //nolint:gosec
```

Comment:
LGTM but I am curious what specifically the linter was complaining about here. Not using https?

Reply:
Here's one of the failures: https://github.com/dagger/dagger/actions/runs/4264916471/jobs/7423612839#step:4:665

```
cmd/engine/debug.go:40:5: G114: Use of net/http serve function that has no support for setting timeouts (gosec)
```

I think it wants us to configure a `ReadHeaderTimeout`; DeepSource has a nice example: https://deepsource.io/directory/analyzers/go/issues/GO-S2114

Seems like a potential DoS vector (opening a ton of requests and never sending request headers). Maybe we care about this for the playground?

Reply:
cc @marcosnils but it's fine here for now I think